doc/uploading-files.md

```diff
@@ -0,0 +1,55 @@
+## Uploading files
+
+Lets say you want your blog posts to contain images.
+Here's how you add file-upload in `data/collection/post.json`:
+
+    "properties":{
+
+      ...
+
+      "picture": {
+          "title": "Picture",
+          "type": "string",
+          "media": {
+              "binaryEncoding": "base64",
+              "type": "image/png"
+          },
+          "links": [
+             {
+               "href": "/custom_endpoint/{{self}}.png",
+               "mediaType": "image/png"
+             }
+          ]
+      }
+
```

Comment:
Uploaded files will then be stored along with the other fields in the collection. Is that right? Maybe add a comment like that here?